src/app/qgisapp.cpp

```diff
@@ -3683,7 +3683,7 @@ void QgisApp::createStatusBar()
   mMagnifierWidget->setObjectName( QStringLiteral( "mMagnifierWidget" ) );
   mMagnifierWidget->setFont( statusBarFont );
   connect( mMapCanvas, &QgsMapCanvas::magnificationChanged, mMagnifierWidget, &QgsStatusBarMagnifierWidget::updateMagnification );
-  connect( mMagnifierWidget, &QgsStatusBarMagnifierWidget::magnificationChanged, mMapCanvas, &QgsMapCanvas::setMagnificationFactor );
+  connect( mMagnifierWidget, &QgsStatusBarMagnifierWidget::magnificationChanged, mMapCanvas, [ = ]( double factor ) { mMapCanvas->setMagnificationFactor( factor ); } );
```

Comment:
why is this change required since there is a default value in the slot?

Reply:
The Qt5 `connect` syntax [does not support default arguments in slots](https://wiki.qt.io/New_Signal_Slot_Syntax#Default_arguments_in_slot), unfortunately. I used a lambda function to work around this, which seems to be the standard way to handle this ([example 1](https://artandlogic.com/2013/09/qt-5-and-c11-lambdas-are-your-friend/), [example 2](https://stackoverflow.com/a/40598273/1456857)).